.github/actions/run-polkadot/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Run Polkadot action
+description: Reusable action to build and run a modified version of Polkadot
+inputs:
+  polkadot-version:
+    description: 'Version tag, e.g. polkadot-v1.7.1'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3.3.0
+      with:
+        path: substrate-tip-bot
+    - uses: actions/checkout@v3.3.0
+      with:
+        repository: paritytech/polkadot-sdk
+        ref: ${{ inputs.polkadot-version }}
+        path: polkadot-sdk
+    - name: Apply Polkadot patches
+      run: git apply ../substrate-tip-bot/polkadot.e2e.patch
+      shell: bash
+      working-directory: polkadot-sdk
+    - name: Restore cached Polkadot build
+      id: polkadot-cache-restore
+      uses: actions/cache/restore@v3
+      with:
+        path: |
+          polkadot-sdk/target/release
+        key: ${{ runner.os }}-${{ inputs.polkadot-version }}-${{ hashFiles('substrate-tip-bot/polkadot.e2e.patch') }}
+    - name: Build Polkadot
+      if: steps.polkadot-cache-restore.outputs.cache-hit != 'true'
+      run: |
+        rustup update nightly
+        rustup update stable
+        rustup target add wasm32-unknown-unknown --toolchain nightly
+        cargo build --release --locked --features=fast-runtime -p polkadot
```

Comment:
I guess the environment should be the same environment/rust version as here ?
https://github.com/paritytech/polkadot-sdk/blob/polkadot-v1.7.2/.github/workflows/fmt-check.yml#L18

> docker.io/paritytech/ci-unified:bullseye-1.75.0-2024-01-22-v20240109